### ISSUE TITLE
Publishing offer not possible due to java.net.SocketTimeoutException

### ISSUE THREAD

**Issue by godfuture** [haveno-dex/haveno#1095]
I tried to add a sell offer, but after long waiting time, I get an exception. 

Haveno Reto 1.0.8

```
Juli-04 23:15:09.821 [JavaFX Application Thread] INFO  h.n.p.n.ConnectionState: We have changed the peerType from INITIAL_DATA_EXCHANGE to PEER as we have received all expected initial data responses at connection with peer Optional[bugc27z7lqjgpwmhbuu3kuwoq2bhailj573r32jm5ydwrcqrwjtblnid.onion:1002]/93fac38d-7341-4188-b568-5005335a5a58. 
Juli-04 23:15:11.565 [JavaFX Application Thread] INFO  h.n.p.n.ConnectionState: We have changed the peerType from INITIAL_DATA_EXCHANGE to PEER as we have received all expected initial data responses at connection with peer Optional[gq2reknmxhmf3lnakkyzflr72kbtkpueui7utk3w7qfziuesbjsnkayd.onion:1002]/9013e6b0-4a69-4597-9cc6-ac576828c392. 
Juli-04 23:15:15.501 [JavaFX Application Thread] INFO  h.n.p.p.g.RequestDataHandler: A timeout occurred at sending getDataRequest:GetUpdatedDataRequest(senderNodeAddress=gvkxe3o4x7adptjxnlnkllolrx2snxkxbtt7suq2fnszmg2jtxbrxcad.onion:9999) on nodeAddress:c66xsfwbequrnytwlp7cz7oirs6q4grxmqyny7afuvc6r4e3cn2ynoqd.onion:1003 
Juli-04 23:15:15.504 [JavaFX Application Thread] INFO  h.n.p.p.g.RequestDataHandler: 

>> We send a GetUpdatedDataRequest to peer bqvt3leh37dxdlkgimnmu56rhlpatfks4h2ap2q2zvuubymmtdwd2wad.onion:9999
 
Juli-04 23:15:15.505 [NetworkNode:SendMessage-to-bqvt3leh37dx...] INFO  h.n.p.p.g.m.GetUpdatedDataRequest: Sending a GetUpdatedDataRequest with 48.06 kB and 2121 excluded key entries. Requesters version=1.0.8 
Juli-04 23:15:15.506 [NetworkNode:SendMessage-to-bqvt3leh37dx...] INFO  h.n.p.p.g.m.GetUpdatedDataRequest: Sending a GetUpdatedDataRequest with 48.06 kB and 2121 excluded key entries. Requesters version=1.0.8 
Juli-04 23:15:17.699 [InputHandler-bqvt3leh37dx...] INFO  h.n.p.p.g.m.GetDataResponse: Received a GetDataResponse with 12,4 kB  
Juli-04 23:15:17.700 [pool-47-thread-1] INFO  h.n.p.p.g.RequestDataHandler: 
#################################################################
Data provided by node: bqvt3leh37dxdlkgimnmu56rhlpatfks4h2ap2q2zvuubymmtdwd2wad.onion:9999
Received 3 instances from a GetUpdatedDataRequest
MailboxStoragePayload: 2 / 9,801 kB
OfferPayload: 1 / 2,56 kB
#################################################################
 
Juli-04 23:15:17.707 [pool-47-thread-1] INFO  h.n.p.s.P2PDataStorage: Processing 3 protectedStorageEntries took 7 ms. 
Juli-04 23:15:17.707 [pool-47-thread-1] INFO  h.n.p.s.P2PDataStorage: Processing 0 persistableNetworkPayloads took 0 ms. 
Juli-04 23:15:17.707 [pool-47-thread-1] INFO  h.n.p.p.g.RequestDataHandler: Processing GetDataResponse took 7 ms 
Juli-04 23:15:17.716 [processMailboxEntry-186] INFO  h.n.p.m.MailboxMessageService: Batch processing of 1 mailbox entries took 9 ms 
Juli-04 23:15:17.723 [processMailboxEntry-574] INFO  h.n.p.m.MailboxMessageService: Batch processing of 1 mailbox entries took 19 ms 
Juli-04 23:15:26.602 [InputHandler-c66xsfwbequr...] INFO  h.n.p.n.Connection: Shut down caused by exception java.net.SocketTimeoutException: Read timed out on connection=Connection{peerAddress=Optional[c66xsfwbequrnytwlp7cz7oirs6q4grxmqyny7afuvc6r4e3cn2ynoqd.onion:1003], connectionState=ConnectionState{,
     peerType=INITIAL_DATA_EXCHANGE,
     numInitialDataRequests=1,
     numInitialDataResponses=0,
     lastInitialDataMsgTimeStamp=1720127486600,
     isSeedNode=true,
     expectedInitialDataResponses=5
}, connectionType=OutboundConnection, uid='5251a9ca-0340-4977-989b-d81c60d3ccac'} 
Juli-04 23:15:28.293 [InputHandler-iuisvssjgd6l...] INFO  h.n.p.n.Connection: Shut down caused by exception java.net.SocketTimeoutException: Read timed out on connection=Connection{peerAddress=Optional[iuisvssjgd6lcjrzskp3sro7uwpdjvk7icbz2usyy4h3dtycfk6cgkqd.onion:9999], connectionState=ConnectionState{,
     peerType=PEER,
     numInitialDataRequests=0,
     numInitialDataResponses=0,
     lastInitialDataMsgTimeStamp=0,
     isSeedNode=false,
     expectedInitialDataResponses=5
}, connectionType=OutboundConnection, uid='4096c585-4e94-4817-9fe9-ff915b06bf66'} 
Juli-04 23:18:27.453 [JavaFX Application Thread] INFO  h.c.u.Profiler: Total memory: 238 MB; Used memory: 62,43 MB; Free memory: 175,57 MB; Max memory: 1 GB; No. of threads: 77 
Juli-04 23:19:14.829 [InputHandler-gq2reknmxhmf...] INFO  h.n.p.n.Connection: Shut down caused by exception java.net.SocketTimeoutException: Read timed out on connection=Connection{peerAddress=Optional[gq2reknmxhmf3lnakkyzflr72kbtkpueui7utk3w7qfziuesbjsnkayd.onion:1002], connectionState=ConnectionState{,
     peerType=PEER,
     numInitialDataRequests=1,
     numInitialDataResponses=1,
     lastInitialDataMsgTimeStamp=1720127476563,
     isSeedNode=true,
     expectedInitialDataResponses=5
}, connectionType=OutboundConnection, uid='9013e6b0-4a69-4597-9cc6-ac576828c392'} 
```

**Comment by woodser** [comment]
Looks like it's having connection problems, maybe because of poor internet speed or Tor could be congested. Should resolve itself by restarting the app, trying another internet connection, or trying again in a bit.

**Comment by godfuture** [comment]
Internet was fine, but after two more attempts it worked.

I always saw "Synced with Monero network (over tor) on block xxxx" (translated by me on the fly from german client). So it did not look like I was disconnected or similar. Should I have seen some disconnect warning in the same time of the issue?

**Comment by woodser** [comment]
> Should I have seen some disconnect warning in the same time of the issue?

Not necessarily, the connection to the Monero node can succeed while P2P connections fail.

**Comment by woodser** [comment]
> but after long waiting time, I get an exception.

What was the exception? It should correctly describe a connection problem or be improved.

**Comment by godfuture** [comment]
> > but after long waiting time, I get an exception.
> 
> What was the exception? It should correctly describe a connection problem or be improved.

Yes, you are right. There was a popup dialog stating connection issues. But there was no strack trace presented to me, only the log file was opened. I thought I fetched the right error log, but I could also upload the whole file.

**Comment by woodser** [comment]
Closing as there were related bug fixes in the last release (1.0.10). Please re-open if you see it again.
